drivers/display/display_st7735r.c

```diff
@@ -0,0 +1,494 @@
+		if (*((uint32_t *)context) == DEVICE_PM_ACTIVE_STATE) {
+			st7735r_exit_sleep(data);
+			data->pm_state = DEVICE_PM_ACTIVE_STATE;
+			ret = 0;
+		} else {
+			st7735r_enter_sleep(data);
+			data->pm_state = DEVICE_PM_LOW_POWER_STATE;
+			ret = 0;
+		}
+		break;
+	case DEVICE_PM_GET_POWER_STATE:
+		*((uint32_t *)context) = data->pm_state;
+		break;
+	default:
+		ret = -EINVAL;
+	}
+
+	if (cb != NULL) {
+		cb(dev, ret, context, arg);
+	}
+	return ret;
+}
+#endif /* CONFIG_DEVICE_POWER_MANAGEMENT */
+
+static const struct display_driver_api st7735r_api = {
+	.blanking_on = st7735r_blanking_on,
+	.blanking_off = st7735r_blanking_off,
+	.write = st7735r_write,
+	.read = st7735r_read,
+	.get_framebuffer = st7735r_get_framebuffer,
+	.set_brightness = st7735r_set_brightness,
+	.set_contrast = st7735r_set_contrast,
+	.get_capabilities = st7735r_get_capabilities,
+	.set_pixel_format = st7735r_set_pixel_format,
+	.set_orientation = st7735r_set_orientation,
+};
+
+
+#define ST7735R_INIT(inst)							\
+	const static struct st7735r_config st7735r_config_ ## inst = {		\
+		.spi_name = DT_INST_BUS_LABEL(inst),				\
+		.cs_name = UTIL_AND(						\
+			DT_INST_SPI_DEV_HAS_CS_GPIOS(inst),			\
+			DT_INST_SPI_DEV_CS_GPIOS_LABEL(inst)),			\
+		.cmd_data.name = DT_INST_GPIO_LABEL(inst, cmd_data_gpios),	\
+		.cmd_data.pin = DT_INST_GPIO_PIN(inst, cmd_data_gpios),		\
+		.cmd_data.flags = DT_INST_GPIO_FLAGS(inst, cmd_data_gpios),	\
+		.reset.name = UTIL_AND(						\
+			DT_NODE_HAS_PROP(inst, reset_gpios),			\
+			DT_INST_GPIO_LABEL(inst, reset_gpios)),			\
+		.reset.pin = UTIL_AND(						\
+			DT_NODE_HAS_PROP(inst, reset_gpios),			\
+			DT_INST_GPIO_PIN(inst, reset_gpios)),			\
+		.reset.flags = UTIL_AND(					\
+			DT_NODE_HAS_PROP(inst, reset_gpios),			\
+			DT_INST_GPIO_FLAGS(inst, reset_gpios)),			\
+		.width = DT_INST_PROP(inst, width),				\
+		.height = DT_INST_PROP(inst, height),				\
+		.mdac = DT_INST_PROP(inst, mdac),				\
+		.colmod = DT_INST_PROP(inst, colmod),				\
+		.caset = DT_INST_PROP(inst, caset),				\
+		.raset = DT_INST_PROP(inst, raset),				\
+		.vmctrl1 = DT_INST_PROP(inst, vmctrl1),				\
+		.invctrl = DT_INST_PROP(inst, invctrl),				\
+		.pwctrl1_param = DT_INST_PROP(inst, pwctrl1_param),		\
+		.pwctrl2_param = DT_INST_PROP(inst, pwctrl2_param),		\
+		.pwctrl3_param = DT_INST_PROP(inst, pwctrl3_param),		\
+		.pwctrl4_param = DT_INST_PROP(inst, pwctrl4_param),		\
+		.pwctrl5_param = DT_INST_PROP(inst, pwctrl5_param),		\
+		.frmctrl1_param = DT_INST_PROP(inst, frmctrl1_param),		\
+		.frmctrl2_param = DT_INST_PROP(inst, frmctrl2_param),		\
+		.frmctrl3_param = DT_INST_PROP(inst, frmctrl3_param),		\
+		.pvgam_param = DT_INST_PROP(inst, pvgam_param),			\
+		.nvgam_param = DT_INST_PROP(inst, nvgam_param),			\
+	};									\
+										\
+	static struct st7735r_data st7735r_data_ ## inst = {			\
+		.config = &st7735r_config_ ## inst,				\
+		.spi_config.slave = DT_INST_REG_ADDR(inst),			\
+		.spi_config.frequency = UTIL_AND(				\
```

Comment:
It looks like some of these should belong to the config structure (they are static constants)

Reply:
Problem is that this struct is defined by the spi API and I have to assign the cs member based on another struct. 
But it might be possible. I'll see what I can do

Reply:
ok, got it!